### PR TITLE
Added special character '@' on valid file name as requested for YANG

### PR DESCRIPTION
**Pull request by lSindra**
As requested on issue: #951

Signed-off-by: Lucas Sindra <lucassindra01@hotmail.com>

### CODE REVIEW

File: packages/workspace/src/browser/workspace-commands.ts [hunk 1]
```diff
@@ -154,7 +154,7 @@ export class WorkspaceCommandContribution implements CommandContribution {
      * @param parent the parent directory's file stat
      */
     protected validateFileName(name: string, parent: FileStat): string {
-        if (!name || !name.match(/^[\w\-. ]+$/)) {
+        if (!name || !name.match(/^[\w\-.@]+$/)) {
```
Comment (akosyakov) [inline packages/workspace/src/browser/workspace-commands.ts]: Could you try to use https://www.npmjs.com/package/valid-filename?

**Comment by lSindra** [comment]
I changed it, now using valid-filename module, but it is a bit too easy on its rules, at least I think. Maybe you would want to try it?

File: package.json [hunk 2]
```diff
@@ -60,5 +60,8 @@
     "dev-packages/*",
     "packages/*",
     "examples/*"
-  ]
-}
\ No newline at end of file
+  ],
```
Comment (akosyakov) [inline package.json]: Please revert changes to this file, the root package should contain only dev dependencies

File: packages/workspace/package.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
     "@theia/core": "^0.3.0",
     "@theia/filesystem": "^0.3.0",
     "@types/fs-extra": "^4.0.2",
-    "fs-extra": "^4.0.2"
+    "fs-extra": "^4.0.2",
+    "valid-filename": "latest"
```
Comment (akosyakov) [inline packages/workspace/package.json]: Please use the caret range as a version

File: packages/workspace/src/browser/workspace-commands.ts [hunk 4]
```diff
@@ -154,7 +154,9 @@ export class WorkspaceCommandContribution implements CommandContribution {
      * @param parent the parent directory's file stat
      */
     protected validateFileName(name: string, parent: FileStat): string {
-        if (!name || !name.match(/^[\w\-. ]+$/)) {
+        const validFilename = require('valid-filename');
```
Comment (akosyakov) [inline packages/workspace/src/browser/workspace-commands.ts]: Do it once at the top of the file

**Pull request merged**